lib/util.js

```diff
@@ -36,83 +26,111 @@ const buildJSONresult = function(err, warn, inf, metadata) {
 };
 
 /**
- * Build an “options” object based on an HTTP query string or a similar object containing options.
- *
- * An example of <code>constraints</code>:
- * <blockquote><pre>{
- *     "required": ["processDocument"],
- *     "forbidden": ["echidnaReady", "bogusParam"],
- *     "allowUnknownParams": true
- * }</pre>/blockquote>
+ * Build a function that builds an “options” object based on certain parameters.
  *
- * @param {Object} params - an HTTP request query, or a similar object.
- * @param {Object} base - (<strong>optional</strong>) a template or “base” object to build from.
- * @param {Object} constraints - (<strong>optional</strong>) an object listing “required” and/or “forbidden” parameters.
+ * @param {Object} profiles - valid profiles.
  *
- * @returns {Object} an “options” object that can be used by Specberus.
- *
- * @throws {Error} if there is an error in the parameters.
+ * @returns {Function} a function that builds an “options” object based on an HTTP query string or a similar object containing options.
  */
 
-const processParams = function(params, base, constraints) {
-    const result = base ? JSON.parse(JSON.stringify(base)): {};
-    var originFound = false;
-    for (var p in params) {
-        if ('url' === p || 'source' === p || 'file' === p || 'document' === p) {
-            // Origins: only one allowed.
-            if (originFound)
-                throw new Error('Only one of parameters {“url”, “source”, “file”, “document”} is allowed');
-            originFound = true;
-            result[p] = params[p];
-        }
-        else if ('profile' === p) {
-            // Profile: if it's a string, load the corresponding object.
-            if (result.hasOwnProperty(p))
-                throw new Error('Parameter “' + p + '” is used more than once');
-            else if ('string' === typeof params[p]) {
-                if (profiles.hasOwnProperty(params[p]))
-                    result[p] = profiles[params[p]];
-                else if ('auto' === params[p])
-                    result[p] = 'auto';
+const processParams = function(profiles) {
```

Comment:
Can you give that `const` a different name so we don't confuse `processParams` and `exports.processParams`?